isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+ * We store the indices in QStrings because we use them as cube attributes when
+ * we process the mosaic and tracking cubes.
+ *
+ * @param ui        The user interface
+ * @param copyBands Indices of the bands to copy over to the mosaic cube
+ * @param trackBand Index of the tracking band
+ */
+void findTrackBand(UserInterface ui, QString &copyBands, QString &trackBand) {
+  Cube inputCube = Cube(ui.GetFileName("FROM"));
+  Pvl *inputLabel = inputCube.label();
+
+  if (inputLabel->hasObject("IsisCube")) {
+    PvlObject &cubeObject = inputLabel->findObject("IsisCube");
+    if (cubeObject.hasGroup("BandBin")) {
+      PvlGroup &bandBinGroup = cubeObject.findGroup("BandBin");
+      PvlKeyword &currentKeyword = bandBinGroup[0];
+      for (int i = 0; i < currentKeyword.size(); i++) {
+        if (currentKeyword[i] != "TRACKING") {
+          copyBands += QString::number(i + 1); // Make it 1 based
+          copyBands += ",";
+        }
+        else {
+          trackBand = QString::number(i + 1);
+        }
+      }
+    }
+  }
+  if (trackBand == "") {
+    QString msg = "The input cube [" + ui.GetFileName("FROM") + "] does not have a tracking band.";
+    msg += " If you want to create a tracking cube, run a mosaic program.";
+    throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+}
+
+
+/**
+ * Creates the mosaic cube by copying the input cube without the tracking band.
+ * Then, removes the tracking table from the mosaic label and adds a group pointing
+ * to the tracking cube.
+ *
+ * @param ui    The user interface
+ * @param bands The indices of the bands that are not the tracking band
+ */
+void createMosaicCube(UserInterface ui, QString bands) {
+  ProcessByLine p;
+  CubeAttributeInput inAtt = CubeAttributeInput("+" + bands);
+  p.SetInputCube(ui.GetFileName("FROM"), inAtt);
+  p.SetOutputCube("TO");
+  p.StartProcess(copyPixels);
+  p.EndProcess();
+
+  Cube mosaicCube;
+  try {
+    mosaicCube.open(ui.GetFileName("TO"),"rw");
+  }
+  catch (IException &e) {
+    throw IException(IException::User,
+                     "Unable to open the file [" + ui.GetFileName("TO") + "] as a cube.",
+                     _FILEINFO_);
+  }
+  Pvl *mosaicLabel = mosaicCube.label();
+
+  if (mosaicLabel->hasObject("Table")) {
+    PvlObject &tableObject = mosaicLabel->findObject("Table");
+    if (tableObject.hasKeyword("Name")) {
+      PvlKeyword &tableName = tableObject.findKeyword("Name", Pvl::Traverse);
+      if (tableName[0] == "InputImages") {
+        mosaicLabel->deleteObject("Table");
+      }
+    }
+  }
+
+  // Add Tracking Group to the mosaic cube
+  PvlGroup trackingGroup = PvlGroup("Tracking");
+  PvlKeyword trackingName = PvlKeyword("Filename");
+  FileName cubeName = FileName(ui.GetFileName("TO"));
+  trackingName.setValue(cubeName.baseName() + "_tracking.cub"); //Strip off path and add _tracking
+  trackingGroup.addKeyword(trackingName);
+  mosaicLabel->addGroup(trackingGroup);
```

Comment:
This will add the tracking group outside of the IsisCube object. Use `cube::putGroup` to add the tracking group.